src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Runtime/CompilerServices/InlineArrayTests.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                inlineArray8[i] = i + 1;
+            }
+            for (int i = 0; i < 8; i++)
+            {
+                Assert.Equal(i + 1, inlineArray8[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray9Test()
+        {
+            InlineArray9<int> inlineArray9 = new InlineArray9<int>();
+            for (int i = 0; i < 9; i++)
+            {
+                inlineArray9[i] = i + 1;
+            }
+            for (int i = 0; i < 9; i++)
+            {
+                Assert.Equal(i + 1, inlineArray9[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray10Test()
+        {
+            InlineArray10<int> inlineArray10 = new InlineArray10<int>();
+            for (int i = 0; i < 10; i++)
+            {
+                inlineArray10[i] = i + 1;
+            }
+            for (int i = 0; i < 10; i++)
+            {
+                Assert.Equal(i + 1, inlineArray10[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray11Test()
+        {
+            InlineArray11<int> inlineArray11 = new InlineArray11<int>();
+            for (int i = 0; i < 11; i++)
+            {
+                inlineArray11[i] = i + 1;
+            }
+            for (int i = 0; i < 11; i++)
+            {
+                Assert.Equal(i + 1, inlineArray11[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray12Test()
+        {
+            InlineArray12<int> inlineArray12 = new InlineArray12<int>();
+            for (int i = 0; i < 12; i++)
+            {
+                inlineArray12[i] = i + 1;
+            }
+            for (int i = 0; i < 12; i++)
+            {
+                Assert.Equal(i + 1, inlineArray12[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray13Test()
+        {
+            InlineArray13<int> inlineArray13 = new InlineArray13<int>();
+            for (int i = 0; i < 13; i++)
+            {
+                inlineArray13[i] = i + 1;
+            }
+            for (int i = 0; i < 13; i++)
+            {
+                Assert.Equal(i + 1, inlineArray13[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray14Test()
+        {
+            InlineArray14<int> inlineArray14 = new InlineArray14<int>();
+            for (int i = 0; i < 14; i++)
+            {
+                inlineArray14[i] = i + 1;
+            }
+            for (int i = 0; i < 14; i++)
+            {
+                Assert.Equal(i + 1, inlineArray14[i]);
+            }
+        }
+
+        [Fact]
+        public void InlineArray15Test()
+        {
+            InlineArray15<int> inlineArray15 = new InlineArray15<int>();
+            for (int i = 0; i < 15; i++)
+            {
+                inlineArray15[i] = i + 1;
+            }
+            for (int i = 0; i < 15; i++)
+            {
+                Assert.Equal(i + 1, inlineArray15[i]);
+            }
```

Comment:
Nit: could save a bit of code by having a span helper like:
```C#
private static void WriteRead(Span<int> span, int expectedLength)
{
    Assert.Equal(expectedLength, span.Length);
    ... // your for loops
}
```
and then each test would just be, e.g.
```C#
InlineArray14<int> inlineArray14 = new();
WriteRead(inlineArray14, 14);
```